components/dialog/dialog-mixin.js

```diff
@@ -458,6 +464,8 @@ export const DialogMixin = superclass => class extends RtlMixin(superclass) {
 			if (flag) {
 				await this.waitForUpdateComplete();
 				await this._updateSize();
+
+				requestAnimationFrame(() => this._checkForFocusableContentElem()); // check if focusable content is present in order to set focusableContentElemPresent to true
```

Comment:
For the case of async content like in the recently added example. This way the dialog content would not have `tabindex="0"` and focus trap would behave consistently with non-async cases.